drivers/sensor/bme680/bme680.c

```diff
@@ -0,0 +1,427 @@
+/* bme680.c - Driver for Bosch Sensortec's BME680 temperature, pressure,
+ * humidity and gas sensor
+ *
+ * https://www.bosch-sensortec.com/bst/products/all_products/bme680
+ */
+
+/*
+ * Copyright (c) 2016, 2017 Intel Corporation
+ * Copyright (c) 2017 IpTronix S.r.l.
+ * Copyright (c) 2018 Bosch Sensortec GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bme680.h"
+#include <gpio.h>
+#include <i2c.h>
+#include <init.h>
+#include <kernel.h>
+#include <misc/byteorder.h>
+#include <misc/__assert.h>
+#include <sensor.h>
+
+static int bme680_reg_read(struct bme680_data *data, u8_t start, u8_t *buf,
+			   int size)
+{
+	return i2c_burst_read(data->i2c_master, data->i2c_slave_addr, start,
+			      buf, size);
+	return 0;
+}
+
+static int bme680_reg_write(struct bme680_data *data, u8_t reg, u8_t val)
+{
+	return i2c_reg_write_byte(data->i2c_master, data->i2c_slave_addr,
+				  reg, val);
+	return 0;
+}
+
+static void bme680_calc_temp(struct bme680_data *data, u32_t adc_temp)
+{
+	s64_t var1, var2, var3;
+
+	var1 = ((s32_t)adc_temp >> 3) - ((s32_t)data->par_t1 << 1);
+	var2 = (var1 * (s32_t)data->par_t2) >> 11;
+	var3 = ((var1 >> 1) * (var1 >> 1)) >> 12;
+	var3 = ((var3) * ((s32_t)data->par_t3 << 4)) >> 14;
+	data->t_fine = var2 + var3;
+	data->calc_temp = ((data->t_fine * 5) + 128) >> 8;
```

Comment:
Yikes, is it possible to make these statements more readable? Is this implementing an equation from the datasheet? Please provide a reference and/or comment that describes the equation, and use more descriptive variable names.

Reply:
It is obfuscated code and details about the underlying math are confidential.

Reply:
This math is described in the datasheet. The link to the product page is at the top of the source file.

Reply:
For each group of equations, please add a comment in the code that notes which section in the datasheet the equations are documented. Please also add a direct link to the datasheet at the top of the source file.

Reply:
The integer versions are not documented in the datasheet yet. When they are, we will add the relevant links and information.

Reply:
I'm not ok merging this until after the documentation becomes available.

Reply:
Is the new datasheet available yet?

Reply:
Yes, but it does not document this in detail, it just shows the code and refers to the GitHub-hosted open source implementation of the driver